### ISSUE TITLE
Implement Chat Scrollback

### ISSUE THREAD

**Issue by misslivirose** [mozilla/hubs#1619]
**Is your feature request related to a problem? Please describe.**
Currently, chat messages in Hubs disappears very quickly. 

**Describe the solution you'd like**
Chat messages should have a brief history stored (on the client) so that users could scroll back some amount and see what had been said earlier on in their session. 

**Describe alternatives you've considered**
An alternative (and more extensive) solution could be to include chat history as some kind of audit log, but I would suggest that for the first implementation, we consider a client-side solution.

**Additional context**
This is particularly desired in events and meetings with a large number of people.

**Comment by gfodor** [comment]
one facet to the design is if we want the local session chat history to contain messages from the entire session, or continue to expire them out after some period. having a expiration period (of say, 5 minutes) seems somewhat nicer since it'll provide a clean contract around message lifetime, and ensure we don't accidentally increase the "stakes" of interaction by inadvertently introducing a worst-case message lifetime of "forever" since an idle client could theoretically be storing messages over the life of the room. another benefit is this would limit the frustration of users who refresh the page and accidentally lose history -- if their expectation was messages last 5 minutes anyway, they aren't losing much compared to the full history.

the downside of doing a fixed expiration this is similar to the current situation, but less severe: users may be surprised or annoyed to see old messages expiring out, even if they are from N minutes ago. (obv this seems less likely to be as common to hit compared to now.)

**Comment by misslivirose** [comment]
I think that expanding to have the last N minutes would be an iterative improvement over what we have now, and as you mention, the longer that is, the less likely it seems that people will have the expectation that it's over the lifetime of the room. I envision we would still suggest bridging rooms to an alternate platform for that case since that aligns to the room <> channel contract.

**Comment by misslivirose** [comment]
There are some complications with designing this across all of the platforms, client-local history could be done. We need to figure out the right design. Perhaps a secondary modal that has the full client history?

**Comment by emclaren** [comment]
Just noting that I've had a couple different community members ask for this feature recently.

**Comment by gfodor** [comment]
Consider a mobile/non-mobile UI split where chat goes into a modal on mobile due to screen issues.

**Comment by emclaren** [comment]
The accessibility group using Hubs wrote to me about the text chat message user experience on screen readers. From our contact with this group:

When using Hubs with a screen reader, a user can hear the new message notification (audible tone), and then use the screen reader to read through the DOM to get to the message. 

However if/when the message disappears it is not available any more to screen reader. This is one warranting more discussion because it would be nice to have the chat messages automatically read out but not if there are too many messages being read at once.  Twitch is currently exploring this problem space and might be interesting to collaborate with on an appropriate design solution. 

![image(8)](https://user-images.githubusercontent.com/6835324/76282422-d47da380-6254-11ea-9014-be89d0968eab.png)

**Comment by misslivirose** [comment]
More feedback: "Chat history, catching things before they disappear is difficult so having an option to see chat history maybe useful, and maybe also mitigate link/info management problem. Though, how ephemeral are chats intended to be?" - from user Tuism on Discord

**Comment by misslivirose** [comment]
Session local - need to figure out UX on mobile (example: put it in dropdown menu on mobile? flip phone into horizontal?)

**Comment by emclaren** [comment]
I see this is being worked on but just wanted to drop in that we received feedback from the most recent XR Access Symposium event, plus from a researcher who did a walk through with someone with a screenreader that the chat disappeared too fast to be able to read it

**Comment by TechThomas** [comment]
I think this is our top screen reader request from A11yVR!

**Comment by misslivirose** [comment]
Now implemented in #3111 as part of the the upcoming UI redesign
